src/Catalog/Helpers/Db2CatalogProjection.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Data;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Utility class to project <see cref="IDataRecord"/>s retrieved by db2catalog
+    /// into a format that is consumable by the catalog writer.
+    /// </summary>
+    public class Db2CatalogProjection
+    {
+        private readonly PackageContentUriBuilder _packageContentUriBuilder;
+
+        public Db2CatalogProjection(PackageContentUriBuilder packageContentUriBuilder)
+        {
+            _packageContentUriBuilder = packageContentUriBuilder ?? throw new ArgumentNullException(nameof(packageContentUriBuilder));
+        }
+
+        public FeedPackageDetails FromDataRecord(IDataRecord dataRecord)
+        {
+            if (dataRecord == null)
+            {
+                throw new ArgumentNullException(nameof(dataRecord));
+            }
+
+            var packageId = dataRecord["Id"].ToString();
+            var normalizedPackageVersion = dataRecord["NormalizedVersion"].ToString();
+            var listed = dataRecord.GetBoolean(dataRecord.GetOrdinal("Listed"));
+            var hideLicenseReport = dataRecord.GetBoolean(dataRecord.GetOrdinal("HideLicenseReport"));
+
+            var packageContentUri = _packageContentUriBuilder.Build(
+                packageId.ToLowerInvariant(),
+                normalizedPackageVersion.ToLowerInvariant());
+
+            return new FeedPackageDetails(
+                packageContentUri,
+                dataRecord.ReadNullableUtcDateTime("Created"),
```

Comment:
`Created` is not nullable

Reply:
Fixed in https://github.com/NuGet/NuGet.Services.Metadata/pull/524/commits/d9f99d7c369333fa52e42b7ef95f5b43ace38063